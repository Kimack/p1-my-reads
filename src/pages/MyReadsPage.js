import React from 'react'
import BookShelfList from '../components/book/BookShelfList.js'
import { Link } from 'react-router-dom'
import Spinner from 'react-spinkit';
import PropTypes from 'prop-types';

class MyReadsPage extends React.Component {
    render() {
        return (
            <div className="list-books">
                <div className="list-books-title">
                    <h1>I Love Books</h1>
                </div>

                {this.props.isLoading ?
                    (
                        <div className="list-books-loading">
                            <Spinner fadeIn="none" name="ball-pulse-sync" color="#60ac5d" />
                        </div>
                    ) :
                    (
                        <BookShelfList
                            shelves={this.props.shelves}
                            books={this.props.books}
                            onBookShelfChange={this.props.onBookShelfChange} />
                    )
                }

                <div className="open-search">
                    <Link to="/search">Add a book</Link>
                </div>
            </div>
        )
    }
}

MyReadsPage.propTypes = {
	/**
	 * Array of current books on any shelf
	 */
    books: PropTypes.array.isRequired,
	/**
	 * Available shelves array
	 */
    shelves: PropTypes.array.isRequired,
    /**
     * Is the data still loading 
     */
    isLoading: PropTypes.bool.isRequired,
	/**
	 * Function that will be called when the shelf is changed
	 */
    onBookShelfChange: PropTypes.func.isRequired
}

export default MyReadsPage;