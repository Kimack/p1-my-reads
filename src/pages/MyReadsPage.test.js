import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import MyReadsPage from './MyReadsPage';

const shelves = [
    {
        id: "currentlyReading",
        title: "Currently Reading"
    },
    {
        id: "wantToRead",
        title: "Want to Read"
    },
    {
        id: "read",
        title: "Read"
    }
]

configure({ adapter: new Adapter() })

describe('book instance is rendered corretly', () => {
    it('renders without crashing', () => {
        const component = <MyReadsPage books={[{}]} shelves={shelves} isLoading={false} onBookShelfChange={() => { }} />;
        expect(shallow(component)).toMatchSnapshot();
    });

    it('renders without crashing when loading', () => {
        const component = <MyReadsPage books={[{}]} shelves={shelves} isLoading={true} onBookShelfChange={() => { }} />;
        expect(shallow(component)).toMatchSnapshot();
    });
});