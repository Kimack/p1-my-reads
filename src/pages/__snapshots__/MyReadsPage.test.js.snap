// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`book instance is rendered corretly renders without crashing 1`] = `
<div
  className="list-books"
>
  <div
    className="list-books-title"
  >
    <h1>
      I Love Books
    </h1>
  </div>
  <BookShelfList
    books={
      Array [
        Object {},
      ]
    }
    onBookShelfChange={[Function]}
    shelves={
      Array [
        Object {
          "id": "currentlyReading",
          "title": "Currently Reading",
        },
        Object {
          "id": "wantToRead",
          "title": "Want to Read",
        },
        Object {
          "id": "read",
          "title": "Read",
        },
      ]
    }
  />
  <div
    className="open-search"
  >
    <Link
      replace={false}
      to="/search"
    >
      Add a book
    </Link>
  </div>
</div>
`;

exports[`book instance is rendered corretly renders without crashing when loading 1`] = `
<div
  className="list-books"
>
  <div
    className="list-books-title"
  >
    <h1>
      I Love Books
    </h1>
  </div>
  <div
    className="list-books-loading"
  >
    <Spinner
      color="#60ac5d"
      fadeIn="none"
      name="ball-pulse-sync"
      noFadeIn={false}
      overrideSpinnerClassName=""
    />
  </div>
  <div
    className="open-search"
  >
    <Link
      replace={false}
      to="/search"
    >
      Add a book
    </Link>
  </div>
</div>
`;
